tvm/src/codegen/opencl/codegen_xocl_host.cc

```diff
@@ -392,60 +381,59 @@ const int bank[MAX_HBM_BANKCOUNT] = {
     stream << "\n  // set device kernel buffer\n";
-    CHECK(op->args.size() == kernel_args.size());
-    for (size_t k = 0; k < kernel_args.size(); k++) {
-      auto arg_name = kernel_args[k];
-      CHECK(arg_map.count(arg_name));
-      if (arg_map[arg_name].stream_type == StreamType::DMA) {
+    for (size_t k = 0; k < op->args.size(); k++) {
+      auto info = args_info[k];
+      if (info.stream_type == StreamType::DMA) {
         PrintIndent();
         stream << "err = kernel.setArg(" << k << ", "
-               << "buffer_" << kernel_args[k] << ");\n";
+               << "buffer_" << info.name << ");\n";
       }
     }
 
-    // migrate memory objects
-    bool first = true;
+    // Migrate memory objects
+    bool first_buffer = true;
     PrintIndent();
     stream << "err = q.enqueueMigrateMemObjects({";
-    for (size_t k = 0; k < kernel_args.size(); k++) {
-      auto arg_name = kernel_args[k];
-      CHECK(arg_map.count(arg_name));
-      if (arg_map[arg_name].stream_type == StreamType::DMA) {
-        if (!first) stream << ", ";
-        stream << "buffer_" << kernel_args[k];
-        first = false;
+    for (size_t k = 0; k < op->args.size(); k++) {
+      auto info = args_info[k];
+      if (info.stream_type == StreamType::DMA) {
+        if (!first_buffer) stream << ", ";
+        stream << "buffer_" << info.name;
+        first_buffer = false;
       }
     }
     stream << "}, 0/*from host*/);\n";
     stream << "  q.finish();\n";
 
-    // set up timer and start execution
+    // Set up timer and start execution
     stream << "\n  // enqueue kernel function\n";
     stream << "  std::chrono::duration<double> kernel_time(0);\n";
     stream
         << "  auto kernel_start = std::chrono::high_resolution_clock::now();\n";
     stream << "  cl::Event event;\n";
     stream << "  err = q.enqueueTask(kernel, NULL, &event);\n\n";
 
-    // initialize write and read stream
-    if (stream_arg_num > 0) {
-      for (size_t k = 0; k < kernel_args.size(); k++) {
-        auto arg_name = kernel_args[k];
-        CHECK(arg_map.count(arg_name));
-        auto arg_info = arg_map.at(arg_name);
-        auto direction = arg_info.target_device;
-        if (arg_info.stream_type == StreamType::DMA) continue;
+    // Initialize write and read stream
+    if (num_of_stream_args > 0) {
+      for (size_t k = 0; k < op->args.size(); k++) {
+        auto info = args_info[k];
+        auto v = op->args[k].as<Variable>();
+        CHECK(v) << "invalid input var";
+        auto shape = var_shape_map_[v];
+        auto arg_name = info.name;
+
+        if (info.stream_type == StreamType::DMA) continue;
 
         // xcl read stream
-        // TODO(hecmay): add non-blocking stream
-        if (direction == DeviceType::devHost) {
+        // TODO(Hecmay): add non-blocking stream
+        if (info.dev_type == DeviceType::devFPGA) {
           stream << "  "
                  << "cl_stream_xfer_req rd_req_" << arg_name << "{0};\n";
```

Comment:
There are many variables defined by us such as `rd_req_`. Instead of hard-coding it here, at least we should create a C variable for that.